### ISSUE TITLE
Crash from empty input and negative choice

### ISSUE THREAD

**Issue by planet36** [theryangeary/choose#38]
```
$ choose --version
choose 1.3.2
$ export RUST_BACKTRACE=full
```

```
$ echo "0 1 2 3 4" | choose 0
0
```

```
$ echo "0 1 2 3 4" | choose -1
4
```

```
$ echo "" | choose 0
```

```
$ echo "" | choose -1
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', src/choice.rs:186:18
stack backtrace:
   0:     0x55784c8bbf00 - std::backtrace_rs::backtrace::libunwind::trace::ha5edb8ba5c6b7a6c
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b/library/std/src/../../backtrace/src/backtrace/libunwind.rs:90:5
   1:     0x55784c8bbf00 - std::backtrace_rs::backtrace::trace_unsynchronized::h0de86d320a827db2
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x55784c8bbf00 - std::sys_common::backtrace::_print_fmt::h97b9ad6f0a1380ff
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b/library/std/src/sys_common/backtrace.rs:67:5
   3:     0x55784c8bbf00 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h14be7eb08f97fe80
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b/library/std/src/sys_common/backtrace.rs:46:22
   4:     0x55784c8d805f - core::fmt::write::h2ca8877d3e0e52de
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b/library/core/src/fmt/mod.rs:1094:17
   5:     0x55784c8b9f55 - std::io::Write::write_fmt::h64f5987220b618f4
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b/library/std/src/io/mod.rs:1584:15
   6:     0x55784c8bde7b - std::sys_common::backtrace::_print::h7f1a4097308f2e0a
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b/library/std/src/sys_common/backtrace.rs:49:5
   7:     0x55784c8bde7b - std::sys_common::backtrace::print::h1f799fc2ca7f5035
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b/library/std/src/sys_common/backtrace.rs:36:9
   8:     0x55784c8bde7b - std::panicking::default_hook::{{closure}}::hf38436e8a3ce1071
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b/library/std/src/panicking.rs:208:50
   9:     0x55784c8bd94d - std::panicking::default_hook::he2f8f3fae11ed1dd
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b/library/std/src/panicking.rs:225:9
  10:     0x55784c8be48d - std::panicking::rust_panic_with_hook::h79a18548bd90c7d4
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b/library/std/src/panicking.rs:591:17
  11:     0x55784c8bdff7 - std::panicking::begin_panic_handler::{{closure}}::h212a72cc08e25126
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b/library/std/src/panicking.rs:495:13
  12:     0x55784c8bc39c - std::sys_common::backtrace::__rust_end_short_backtrace::hbd6897dd42bc0fcd
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b/library/std/src/sys_common/backtrace.rs:141:18
  13:     0x55784c8bdf89 - rust_begin_unwind
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b/library/std/src/panicking.rs:493:5
  14:     0x55784c7c7cf1 - core::panicking::panic_fmt::h77ecd04e9b1dd84d
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b/library/core/src/panicking.rs:92:14
  15:     0x55784c7c7c3d - core::panicking::panic::h60569d8a39169222
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b/library/core/src/panicking.rs:50:5
  16:     0x55784c7cadb9 - choose::choice::Choice::print_choice::h9aad0eebfbda9462
  17:     0x55784c7d22a4 - choose::main::h34b573cbb97faa1e
  18:     0x55784c7ce263 - std::sys_common::backtrace::__rust_begin_short_backtrace::h58c37d283637d79b
  19:     0x55784c7ce279 - std::rt::lang_start::{{closure}}::h1dfc6e7cf9a3fc13
  20:     0x55784c8be88a - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::hc4354216bf39217c
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b/library/core/src/ops/function.rs:259:13
  21:     0x55784c8be88a - std::panicking::try::do_call::hb68eb312780385cf
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b/library/std/src/panicking.rs:379:40
  22:     0x55784c8be88a - std::panicking::try::h22b8e08595060b8b
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b/library/std/src/panicking.rs:343:19
  23:     0x55784c8be88a - std::panic::catch_unwind::hc64f1a6a0e71b1fc
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b/library/std/src/panic.rs:431:14
  24:     0x55784c8be88a - std::rt::lang_start_internal::h4461fc58637f04f8
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b/library/std/src/rt.rs:34:21
  25:     0x55784c7d2a62 - main
  26:     0x7f496544db25 - __libc_start_main
  27:     0x55784c7c840e - _start
  28:                0x0 - <unknown>
```

**Comment by theryangeary** [comment]
Thanks for reporting, I'll look into this and post a fix once I resolve it.

**Comment by theryangeary** [comment]
Hi @planet36, I've looked into your bug and I believe I have resolved the issue. I've added some testing around the scenario you presented and other similar possible situations. The code is currently on the branch [`38-unwrap-on-empty-choice`](https://github.com/theryangeary/choose/tree/38-unwrap-on-empty-choice). If you would check it out and build it ([instructions](https://github.com/theryangeary/choose/tree/38-unwrap-on-empty-choice#installing-from-source) - don't forget to check out `38-unwrap-on-empty-choice`) and let me know if you see anything I missed, I would greatly appreciate it. Thanks again for reporting this issue!
